src/ensemble_scheduler.cc

```diff
@@ -1264,6 +1264,11 @@ EnsembleContext::ScheduleSteps(
       }
     }
     if (should_schedule) {
+      // If the ensemble request is cancelled, propagate the cancellation to the
+      // next request step.
+      if (context->request_tracker_->Request()->IsCancelled()) {
+        step->request_->Cancel();
```

Comment:
I think rather than propagating the cancellation status we should stop scheduling new requests altogether. I believe for errors ensemble wouldn't call the next steps and I think we should do something similar here.

Reply:
This is a good point, but there is a corner case. If the next step is a request belonging to an in-flight sequence, the entire in-flight sequence should be cancelled along with any queued requests on that sequence that may not have been explicitly cancelled. The ensemble scheduler is not knowledgeable enough to perform the cancellation, so it need to be deferred to the sequence scheduler on the next step. The simplest way of passing the information of cancellation to the sequence scheduler is to simply mark the next request as cancelled and pass it downstream.

Reply:
thanks for the detailed response! Resolving.